src/app/features/courses/courses.component.html

```diff
@@ -4,8 +4,9 @@
 </ng-container>
 
 <ng-template #courseList>
-  <ng-container *ngIf="courses.length>0; else emptyList" > 
-    <app-courses-list (showCourse)="handleShowCourseInfo($event)" [editable]="true" [courses]="courses"></app-courses-list>
+  <ng-container *ngIf="courses.length>0; else emptyList" >
+    <app-search (search)="handleSearch($event)" [placeholder]="'Course...'"></app-search> 
+    <app-courses-list (showCourse)="handleShowCourseInfo($event)" [editable]="true" [courses]="searchedCourses.length > 0 ? searchedCourses : courses"></app-courses-list>
```

Comment:
This looks like sg that needs to be on the controller level, not the view.
For `[courses]` input on CourseListComponent, you shoul opt into a solution where it is just reading the content from the  controller. For example: `[courses]="getMyFilteredCourses()"`
The main idea is to move business logic to the controller (.ts)

Reply:
SearchBar functionality was not really a requirement it was just a pre-version, but refined now for a better experience.
Done